contracts/SemiFungiblePositionManager.sol

```diff
@@ -888,53 +888,42 @@ contract SemiFungiblePositionManager is ERC1155, Multicall {
 
-                /// we need to compare (deltaX = itm0 - itm1/price) to (deltaY =  itm1 - itm0 * price) to only swap the owed balance
-                /// To reduce the number of computation steps, we do the following:
-                ///    deltaX = (itm0*sqrtPrice - itm1/sqrtPrice)/sqrtPrice
-                ///    deltaY = -(itm0*sqrtPrice - itm1/sqrtPrice)*sqrtPrice
-
-                int256 net0 = itm0 + PanopticMath.convert1to0(itm1, sqrtPriceX96);
-
-                int256 net1 = itm1 + PanopticMath.convert0to1(itm0, sqrtPriceX96);
-
-                // if net1 is negative, then the protocol has a surplus of token0
-                zeroForOne = net1 < net0;
+                // implement a single "netting" swap. Thank you @danrobinson for this puzzle/idea
+                // note: negative ITM amounts denote a surplus of tokens (burning liquidity), while positive amounts denote a shortage of tokens (minting liquidity)
+                // compute the approximate delta of token0 that should be resolved in the swap at the current tick
+                // we do this by flipping the signs on the token1 ITM amount converting+deducting it against the token0 ITM amount
+                // couple examples (price = 2 1/0):
+                //  - 100 surplus 0, 100 surplus 1 (itm0 = -100, itm1 = -100)
+                //    normal swap 0: 100 0 => 200 1
+                //    normal swap 1: 100 1 => 50 0
+                //    final swap amounts: 50 0 => 100 1
+                //    netting swap: net0 = -100 - (-100/2) = -50, ZF1 = true, 50 0 => 100 1
+                // - 100 surplus 0, 100 shortage 1 (itm0 = -100, itm1 = 100)
+                //    normal swap 0: 100 0 => 200 1
+                //    normal swap 1: 50 0 => 100 1
+                //    final swap amounts: 150 0 => 300 1
+                //    netting swap: net0 = -100 - (100/2) = -150, ZF1 = true, 150 0 => 300 1
+                // - 100 shortage 0, 100 surplus 1 (itm0 = 100, itm1 = -100)
+                //    normal swap 0: 200 1 => 100 0
+                //    normal swap 1: 100 1 => 50 0
+                //    final swap amounts: 300 1 => 150 0
+                //    netting swap: net0 = 100 - (-100/2) = 150, ZF1 = false, 300 1 => 150 0
+                // - 100 shortage 0, 100 shortage 1 (itm0 = 100, itm1 = 100)
+                //    normal swap 0: 200 1 => 100 0
+                //    normal swap 1: 50 0 => 100 1
+                //    final swap amounts: 100 1 => 50 0
+                //    netting swap: net0 = 100 - (100/2) = 50, ZF1 = false, 100 1 => 50 0
+                // - = Net surplus of token0
+                // + = Net shortage of token0
+                int256 net0 = itm0 - PanopticMath.convert1to0(itm1, sqrtPriceX96);
+
+                zeroForOne = net0 < 0;
 
                 //compute the swap amount, set as positive (exact input)
-                swapAmount = zeroForOne ? net0 : net1;
+                swapAmount = -net0;
```

Comment:
>We always want swapAmount to always be negative so that we know exactly how many tokens we'll get through a exactOutput swap.
This is true when there is only one net ITM amount, because we want to ensure the user only has to pay in their specified token. 

However, in this case, the user will be paying in both tokens regardless. Since this swap is an approximation anyway, it is irrelevant whether we use exact input or exact output. If we change the case `net0<0` to exact output as you suggested,  the input would simply slip instead. There's no preference for the accuracy of the input vs the output when we have ITM amounts consisting of both tokens.

Regardless of what amounts actually get swapped here, the user will pay the correct amount. The only concerns as to accuracy are
a) If only one ITM amount is present, is the user able to pay in only that token, or are they forced to handle some dust in the other token?
b) In the case of the netting swap (a. working correctly precludes an unreasonable swap amount in its case), is the swap amount *reasonably* close to the outcome if there are two swaps and is it reasonable for the user to execute/pay for (i.e on certain pools the previous logic would create massive swaps at burn that would be impossible to actually execute)
Both of which are satisfied with this logic